src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/JmsLocalStackConfig.kt

```diff
@@ -12,19 +12,30 @@ import org.springframework.beans.factory.annotation.Value
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import uk.gov.justice.hmpps.sqs.HmppsQueue
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
 
 @Configuration
 @ConditionalOnProperty(name = ["sqs.provider"], havingValue = "localstack")
-class JmsLocalStackConfig {
+class JmsLocalStackConfig(private val hmppsQueueService: HmppsQueueService) {
+
   @Bean("awsSqsClient")
   fun awsSqsClientLocalstack(
     @Value("\${sqs.endpoint.url}") serviceEndpoint: String,
-    @Value("\${sqs.endpoint.region}") region: String
+    @Value("\${sqs.endpoint.region}") region: String,
+    @Value("\${sqs.queue.name}") queueName: String,
+    awsSqsDlqClient: AmazonSQS,
+    @Value("\${sqs.dlq.name}") dlqName: String,
   ): AmazonSQS =
     AmazonSQSClientBuilder.standard()
       .withEndpointConfiguration(AwsClientBuilder.EndpointConfiguration(serviceEndpoint, region))
       .withCredentials(AWSStaticCredentialsProvider(AnonymousAWSCredentials()))
       .build()
+      .also {
+        hmppsQueueService.registerHmppsQueue(
+          HmppsQueue(it, queueName, awsSqsDlqClient, dlqName)
```

Comment:
I know I haven't commented on this before (sorry), but this method looks odd on the queue service.  In kotlin land seems odd to have to call the constructor on a queue and then register it.  Might be nicer to just call the queue (by parameter names):
```
hmppsQueueService.registerHmppsQueue(client = it, queue = queueName, dlqClient = awsSqsDlqClient, dlqQueue = dlqName)
```